scwidgets/widgets.py

```diff
@@ -0,0 +1,230 @@
+                                   value='',
+                                   continuous_update=False)
+
+        self.button = widgets.Button(description=name)
+        self.button.on_click(self._on_button_clicked)
+
+        self.output_area = widgets.Output()
+
+        self.children = [
+            widgets.VBox([
+                widgets.HBox(self.input_widgets + [self.output, self.button]),
+                self.output_area
+            ])
+        ]
+
+        self.subscribers = []
+
+    def _setup_input_widgets(self, inputs, descriptions, options):
+        """
+        Creates a Combobox widget for each entry in the inputs dict.
+        The placeholder is set based on the descriptions dict and the
+        options are set based on the options dict.
+        """
+        for name in inputs.keys():
+            placeholder = descriptions[name] if name in descriptions else name
+            option = options[name] if name in options else []
+            option = option() if callable(option) else option
+            self.input_widgets.append(
+                widgets.Combobox(placeholder=placeholder,
+                                 continuous_update=False,
+                                 options=option))
+
+    def _retrive_kwargs(self):
+        kwargs = {
+            name: converter(item.value)
+            for name, converter, item in zip(
+                self.inputs.keys(), self.inputs.values(), self.input_widgets)
+        }
+        return kwargs
+
+    def _on_button_clicked(self, button):
+        self.output_area.clear_output()
+        with self.output_area:
+            if self.output.value:
+                output_name = self.output.value
+            else:
+                print('Invalid inputs: No output name specified')
+                return
+            self.scope[output_name] = self.process()
+
+    def process(self):
+        """
+        Calls the wrapped function using the
+        parameter values specified.
+
+        Returns: Output of the wrapped callable.
+        """
+        try:
+            kwargs = self._retrive_kwargs()
+        except ValueError as e:
+            print(f'Invalid inputs: {e}')
+            return
+
+        return self.callable(**kwargs)
+
+
+def _repr_html_(scope):
+    """
+    Helper method to display the scipp objects in a given scope.
+    """
+    from IPython import get_ipython
+    ipython = get_ipython()
+    out = ''
+    for category in ['Variable', 'DataArray', 'Dataset']:
+        names = ipython.magic(f"who_ls {category}")
+        out += f"<details open=\"open\"><summary>{category}s:"\
+               f"({len(names)})</summary>"
+        for name in names:
+            html = sc.table_html.make_html(eval(name, scope))
+            out += f"<details style=\"padding-left:2em\"><summary>"\
+                   f"{name}</summary>{html}</details>"
+        out += "</details>"
+    display(HTML(out))
+
+
+class PlotWidget(widgets.Box):
```

Comment:
I wonder whether the plot widget is actually just a special case of something more generic? Is it simply calling a function that has no name assigned to its return value? Maybe we should initially just support that, if it does mostly the same? The object list with auto updates feels too specialized and brittle for the initial version of this package.

Reply:
Yes some form of display widget should work which displays the output of the wrapped function rather than returning it to scope.